server/conf/evolutions/default/23.sql

```diff
@@ -0,0 +1,23 @@
+
+# --- !Ups
+CREATE TYPE REWARD_TYPE AS ENUM (
+  'PoW',
+  'PoS',
+  'TPoS_OWNER',
+  'TPoS_MERCHANT'
+);
+
+CREATE TABLE block_rewards(
+  blockhash HASH_TYPE NOT NULL,
+  value AMOUNT_TYPE NOT NULL,
+  address ADDRESS_TYPE NOT NULL,
+  type REWARD_TYPE NULL,
+  -- constraints
+  CONSTRAINT block_rewards_pk PRIMARY KEY (blockhash, address)
```

Comment:
Some observations:
- It makes more sense for the primary key to be composed on the block and the reward type.
- The reward type must be not null.
- Create an index for the address, we'll end up retrieving rewards for a given address (use `CREATE INDEX CONCURRENTLY ...`).
- Add a foreign key constraint to the blocks table.

Reply:
Any reason for creating the index concurrently?, the table was just created and will be empty anyway.
Also the evolution fails with this error `ERROR: CREATE INDEX CONCURRENTLY cannot run inside a transaction block` if i use `CREATE INDEX CONCURRENTLY`

Reply:
You are right, forget about the `CONCURRENTLY`.